### PR TITLE
Amend tests to include all codecs

### PR DESCRIPTION
**Pull request by sarthakaggarwal97**
### Description
Adds additional codecs to the tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java [hunk 1]
```diff
@@ -427,7 +427,16 @@ protected void randomIndexTemplate() {
...
             if (annotation != null && annotation.value().length == 1 && "*".equals(annotation.value()[0])) {
-                randomSettingsBuilder.put("index.codec", randomFrom(CodecService.DEFAULT_CODEC, CodecService.BEST_COMPRESSION_CODEC));
+                randomSettingsBuilder.put(
+                    "index.codec",
+                    randomFrom(
+                        CodecService.DEFAULT_CODEC,
+                        CodecService.BEST_COMPRESSION_CODEC,
+                        CodecService.LUCENE_DEFAULT_CODEC,
+                        CodecService.ZSTD_CODEC,
+                        CodecService.ZSTD_NO_DICT_CODEC
```
Comment (shwetathareja) [inline test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java:437]: create a static list in this base class and use it all the other integ test classes

File: test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java [hunk 2]
```diff
@@ -364,6 +364,14 @@ public abstract class OpenSearchIntegTestCase extends OpenSearchTestCase {
     private static OpenSearchIntegTestCase INSTANCE = null; // see @SuiteScope
     private static Long SUITE_SEED = null;
 
+    public static List<String> codecs = List.of(
```
Comment (shwetathareja) [inline test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java:367]: [minor]: CODECS in caps for static field.

Comment (sarthakaggarwal97) [reply test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java]: ah right, dont know I missed this! thankss. I've updated!

**Review by Poojita-Raj — APPROVED** [review]

**Comment by Poojita-Raj** [comment]
There are 3 simultaneous PRs around the new codec introductions that I see: #8737, #8745 and this one (#8907). Is there a meta issue detailing why these changes are being made? 

I think they should all be linked to a meta to let reviewers see the larger context and the order in which these PRs should be reviewed and merged in.

**Comment by sarthakaggarwal97** [comment]
@Poojita-Raj these three PR's are majorly for refactoring few things related to compression codecs and their settings. All of which are independent of each other. I don't think a meta issue was required for such refactors.

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**